docs/remote_debugging.md

```diff
@@ -0,0 +1,218 @@
+require 'debug/session'
+DEBUGGER__.open
+
+# your application code
+```
+
+## TCP/IP
+
+If you want to use TCP/IP for remote debugging, you need to specify the port and host with `--port` like
+
+```
+rdbg --open --port 12345 # binds to 127.0.0.1:12345
+rdbg --open --port 12345 --host myhost.dev # binds to myhost.dev:12345
+```
+
+If you don't use the `rdbg` executable, you can also specify port and host with `RUBY_DEBUG_PORT` and `RUBY_DEBUG_HOST` environment variables:
+
+```shell
+$ RUBY_DEBUG_PORT=12345 RUBY_DEBUG_HOST=localhost ruby target.rb
+```
+
+# Connect To A Remote Debuggee
+
+## Debugger Console
+
+You can connect to the debuggee with `rdbg --attach` command (`rdbg -A` for short).
+
+```shell
+$ rdbg --attach
+[1, 7] in target.rb
+=>    1| a = 1
+      2| b = 2
+      3| c = 3
+      4| d = 4
+      5| p [a, b, c, d]
+      6|
+      7| __END__
+=>#0    <main> at target.rb:1
+
+(rdbg:remote)
+```
+
+If there is only one opened UNIX domain sockets in the socket directory, `rdbg --attach` will connect to it automatically.
+
+If there are more than one socket files, you need to specify the socket name after the `--attach` flag:
+
+```
+❯ rdbg --attach
+Please select a debug session:
+  ruby-debug-st0012-64507
+  ruby-debug-st0012-68793
+
+❯ rdbg --attach ruby-debug-st0012-64507
+[1, 1] in target.rb
+=>   1| a = 1
+=>#0    <main> at target.rb:1
+(rdbg:remote)
+```
+
+When `rdbg --attach` connects to the debuggee, you can use any debugger commands like in local debugger console. When a debuggee program exits, the remote console will also terminate.
+
+NOTE: If you use the `quit` command, only the remote console exits and the debuggee program continues to run (and you can connect to it again). If you want to exit the debuggee program as well, use `kill` command instead.
+
+### Through TCP/IP
+
+To connect to the debuggee via TCP/IP, you need to specify the port.
+
+```shell
+$ rdbg --attach 12345
+```
+
+If you want to choose the host to bind, you can use `--host` option.
+Note that all messages between the debugger and the debuggee are **NOT** encrypted. So please use remote debugging carefully.
+
+## VSCode
+
+Like other languages, you can use this debugger with the VSCode. And there are multiple ways to do it as long as you have the extension installed:
+
+- [VSCode rdbg Ruby Debugger - Visual Studio Marketplace](https://marketplace.visualstudio.com/items?itemName=KoichiSasada.vscode-rdbg) (`v0.0.9` or later is required)
+
+You can configure the extension in `.vscode/launch.json`. Please see the extension page for more details.
+
+You can also check [Debugging in Visual Studio Code](https://code.visualstudio.com/docs/editor/debugging) for more information about VSCode's debugger features.
+
+### Start The Debuggee Program From VSCode
+
+1. Open a `.rb` file (e.g. `target.rb`)
+1. Register breakpoints with "Toggle breakpoint" in the `Run` menu (or press F9)
+1. Choose "Debug current file with rdbg" in the `Run` menu (or press F5)
```

Comment:
<img width="277" alt="Screen Shot 2022-07-04 at 5 17 12 PM" src="https://user-images.githubusercontent.com/15860699/177217258-82e2cf5a-14b4-4590-b1b2-ec884d504a4b.png">

I don't see a "Debug current file with rdbg".

Reply:
I misunderstood what the `Run` menu means here and thought it's the `Run and Debug` panel. Will fix it.